Objects/interpolationobject.c

```diff
@@ -0,0 +1,217 @@
+/* t-string Interpolation object implementation */
+
+#include "Python.h"
+#include "pycore_initconfig.h"      // _PyStatus_OK
+#include "pycore_interpolation.h"
+
+static int
+_conversion_converter(PyObject *arg, PyObject **conversion)
+{
+    if (arg == Py_None) {
+        return 1;
+    }
+
+    if (!PyUnicode_Check(arg)) {
```

Comment:
Should it be a strict Unicode or is a subclass of strings allowed here?

Reply:
The runtime will never create an `Interpolation` with a str subclass, but it shouldn't be a problem for the check to be more relaxed here.